ext/XS-APItest/APItest.xs

```diff
@@ -6802,8 +6802,14 @@ test_Gconvert(SV * number, SV * num_digits)
         int len;
     CODE:
         len = (int) SvIV(num_digits);
-        if (len > 99) croak("Too long a number for test_Gconvert");
-        if (len < 0) croak("Too short a number for test_Gconvert");
+        if (sizeof(NV) == 8) {
+            if(len > 92)
+              croak("Too long a number for test_Gconvert");
+        }
+        else if (len > 91) /* exponent can be 1 digit larger */
+            croak("Too long a number for test_Gconvert");
```

Comment:
It would be worth a comment to explain these limits; I think something like the below would be sufficient:
```
  /* bufsize - float_need = 127 - 35 = 92. See comments around the setting of float_need in sv_vcatpvfn_flags() */
```

Reply:
This is a little bit different to what happened in sv.c.

In sv.c the buffersize was 127 and we said we'll write to the buffer only if the no. of digits is less than buffersize - 35.
In APItest.xs, the buffersize is 100 and we would write to the buffer so long as the no. of digits is less than 100 - 1, which allowed only for the terminating NULL byte. The overflow warning was therefore justified.

All I've done is to allow for a total of 8 bytes - one for the terminating null, one for the radix point, one for a leading '-', two for an 'e-' and three for the exponent digits.
For the long double, I've allowed an extra byte because there can be four exponent digits.
So, it's a case of 100 - 8 = 92 and 100 - 9 = 91.

That was enough to suppress the overflow warning.
If you think I should allow more than those 8 (or 9) bytes just specify that number and I'll accept it without question.
Also, whatever the no. of bytes we decide to allow, I will provide comments that explain just how we arrived at that number.

(BTW, I'll modify my sprintf2.t comments to point to https://github.com/Perl/perl5/pull/18262 , as you've suggested).

Reply:
I don't have a problem with the number, as long as there's some explanation of where it comes from.